src/lib/filter.ts

```diff
@@ -0,0 +1,6 @@
+export const filterObject = (
+  targetObject: object,
+  predicate: (entry: [string, any]) => boolean,
+): object => {
+  return Object.fromEntries(Object.entries(targetObject).filter(predicate));
+};
```

Comment:
`object` 랑 `any` 둘다 안좋은 타입이라 이건 어때여

```suggestion
export const filterObject = <T extends Record<string, unknown>>(
  targetObject: T,
  predicate: (entry: [string, T[keyof T]]) => boolean,
): Partial<T> => {
  return Object.fromEntries(Object.entries(targetObject).filter(predicate));
};
```

Reply:
Object.fromEntries(targetObject)의 타입이 [string, unknown][]으로 떠서 [string, T[keyof T]][]랑 불일치 한다고 에러가 나는디, 반환값도 Record<string, unknown>으로 나와서 Partial<T>랑 불일치한다고 나오고. 코드에는 문제 없으니 그냥 type assertion 해주면 되나? 다른 해결 방법 있나 해서...
아니면 type safe한 entry 함수를 만들어서 써야 하나?

타스 버전은 5.0.2임당

Reply:
@designDefined `Object.entries` 의 한계라 그냥 assertion 해줘야될듯 ㅜㅜ